src/etl/ETLHelpers.h

```diff
@@ -89,10 +98,13 @@ class NetworkValidatedLedgers
     }
 };
 
-/// Generic thread-safe queue with an optional maximum size
-/// Note, we can't use a lockfree queue here, since we need the ability to wait
-/// for an element to be added or removed from the queue. These waits are
-/// blocking calls.
+// TODO: does the note make sense? lockfree queues provide the same blocking behaviour just without mutex, don't they?
```

Comment:
i don't quite understand this. lockfree queue can not wait?

Reply:
That's what the old comment was suggesting. It's worth checking but i think lockfree has nothing to do with waiting or not - it's just a way to implement the same async queue but using `compare_exchange_*` and without a `mutex`.